interfaces/cython/cantera/func1.pyx

```diff
@@ -63,6 +63,9 @@ cdef class Func1:
 
     def __init__(self, c):
         if hasattr(c, '__call__'):
+            if type(c).__name__ == 'method':
+                raise TypeError("Unable to use callback from 'method'.")
+
```

Comment:
What's the problem here? Using bound member functions as `Func1` objects works just fine, to the best of my knowledge.

Reply:
Have a look at the `interfaces/cython/cantera/test/test_reaction.py` (specifically `TestCustom::setUp`). For whatever reason the type is modified when you assign `Func1` to a static member variable.

Reply:
If I'm correctly interpreting what you had initially tried in that test correctly, it was this, right?
```python
class TestCustom(TestElementary):
    _rate = lambda T: 38.7 * T**2.7 * exp(-3150.15428/T)
```
The resulting behavior and error is indeed a somewhat surprising, but I guess it makes sense if you consider that this is basically the equivalent of writing
```python
class TestCustom(TestElementary):
    def _rate(T):
        return 38.7 * T**2.7 * exp(-3150.15428/T)
```
where it's more clear that the method ought to have a `self` parameter, or be declared with the `@staticmethod` decorator.

Disabling the use of `method` for `Func1` as above prevents the following use case, which I think we should still allow:
```python
class MyRate:
    def __init__(self, A, b, Ta):
        self.A = A
        self.b = b
        self.Ta = Ta
    def eval(self, T):
        return self.A * T ** self.b * np.exp(-self.Ta/T)

rate = MyRate(2e8, 0.1, 2100)
rxn = ct.CustomReaction(equation=..., rate=rate.eval, kinetics=gas)
```